core/trino-main/src/test/java/io/trino/operator/output/TestPositionsAppender.java

```diff
@@ -463,4 +509,290 @@ public void appendTo(PositionsAppender positionsAppender)
+        }
+
+        @Override
+        public BlockBuilder createBlockBuilder(BlockBuilderStatus blockBuilderStatus, int expectedEntries)
+        {
+            return createBlockBuilder(blockBuilderStatus, expectedEntries, 32);
+        }
+
+        @Override
+        public Object getObjectValue(ConnectorSession session, Block block, int position)
+        {
+            if (block.isNull(position)) {
+                return null;
+            }
+
+            Slice slice = block.getSlice(position, 0, block.getSliceLength(position));
+            return slice.toStringUtf8();
+        }
+
+        @Override
+        public void appendTo(Block block, int position, BlockBuilder blockBuilder)
+        {
+            if (block.isNull(position)) {
+                blockBuilder.appendNull();
+            }
+            else {
+                block.writeBytesTo(position, 0, block.getSliceLength(position), blockBuilder);
+                blockBuilder.closeEntry();
+            }
+        }
+
+        public void writeString(BlockBuilder blockBuilder, String value)
+        {
+            writeSlice(blockBuilder, Slices.utf8Slice(value));
+        }
+
+        @Override
+        public void writeSlice(BlockBuilder blockBuilder, Slice value)
+        {
+            writeSlice(blockBuilder, value, 0, value.length());
+        }
+
+        @Override
+        public void writeSlice(BlockBuilder blockBuilder, Slice value, int offset, int length)
+        {
+            blockBuilder.writeBytes(value, offset, length).closeEntry();
+        }
+
+        @Override
+        public Slice getSlice(Block block, int position)
+        {
+            return block.getSlice(position, 0, block.getSliceLength(position));
+        }
+
+        @Override
+        public boolean isComparable()
+        {
+            return true;
+        }
+
+        @Override
+        public TypeOperatorDeclaration getTypeOperatorDeclaration(TypeOperators typeOperators)
+        {
+            return TYPE_OPERATOR_DECLARATION;
+        }
+
+        @ScalarOperator(EQUAL)
+        private static boolean equalOperator(@BlockPosition Block leftBlock, @BlockIndex int leftPosition, @BlockPosition Block rightBlock, @BlockIndex int rightPosition)
+        {
+            int leftLength = leftBlock.getSliceLength(leftPosition);
+            int rightLength = rightBlock.getSliceLength(rightPosition);
+            if (leftLength != rightLength) {
+                return false;
+            }
+            return leftBlock.equals(leftPosition, 0, rightBlock, rightPosition, 0, leftLength);
+        }
+
+        @ScalarOperator(XX_HASH_64)
+        private static long xxHash64Operator(@BlockPosition Block block, @BlockIndex int position)
+        {
+            return block.hash(position, 0, block.getSliceLength(position));
+        }
+    }
+
+    private static class TestVariableWidthBlockBuilder
```

Comment:
Do we need to have a special builder? Maybe you could just add some kind of adaptation that would convert `VariableWidthBlock` to `TestVariableWidthBlock`?

Reply:
I added adaptation instead of Type and Builder